### PR TITLE
Add activity diagrams to DG under Implementation

### PR DESCRIPTION
**Pull request by ruiyigan**
Please ignore the file change! The files I like for review are commented below. (TLDR I made a mistake merging the previous approved PR and committing)

**Add activity diagrams to show logic of different modes**
CommandActivityFlowDiagram

CommandValidityActivityDiagram

### CODE REVIEW

File: docs/images/CommandActivityFlowDiagram.png
Comment (rockman007372) [inline docs/images/CommandActivityFlowDiagram.png:1]: Hmmm I wonder what the node Mode is - I havent seen this notation before... 

Also should we have 3 "Check validity of command"? After we check which more we are in we all do the check validity right?

File: docs/images/ModelClassDiagram.png
Comment (rockman007372) [inline docs/images/ModelClassDiagram.png:1]: Looks correct to me! Maybe we can change the layout to make it looks clearer in the future.

**Review by rockman007372 — APPROVED** [review]
I added some comments  regarding the activity diagram. Beyond that lgtm

File: docs/images/CommandActivityFlowDiagram.png
Comment (ruiyigan) [reply docs/images/CommandActivityFlowDiagram.png:1]: it is actually a new feature for PlantUML: https://plantuml.com/activity-diagram-beta for enums.

Yea. I put 3 different ones as the validity is different depending on the modes. Hmm what do the rest think? @unfazing @junianob @HJ-PRK

**Pull request merged**